### ISSUE TITLE
No buttons for NodOn

### ISSUE THREAD

**Issue by bogri** [Z-Wave-Me/z-way-issues#117]
I've managed to include the device NodOn [http://nodon.fr/en/z-wave/z-wave-soft-remote_8-2] and interview is fine, but I don't get any buttons so I can configure scenes. Pressing device buttons are triggering events to the controller - v2.2.2. Attaching the log during inclusion.
[nodOnInclusion.txt](https://github.com/Z-Wave-Me/z-way-issues/files/540108/nodOnInclusion.txt)

**Comment by bogri** [comment]
This is what the log says when a button is pressed:

[2016-10-22 18:38:10.633] [D] [zway] RECEIVED: ( 01 09 00 04 00 18 03 20 01 FF 37 )
[2016-10-22 18:38:10.634] [D] [zway] SENT ACK
[2016-10-22 18:38:10.634] [D] [zway] SETDATA devices.24.data.lastReceived = 0 (0x00000000)
[2016-10-22 18:38:10.634] [D] [zway] SETDATA devices.1.instances.0.commandClasses.32.data.srcNodeId = 24 (0x00000018)
[2016-10-22 18:38:10.634] [D] [zway] SETDATA devices.1.instances.0.commandClasses.32.data.srcInstanceId = 0 (0x00000000)
[2016-10-22 18:38:10.634] [D] [zway] SETDATA devices.1.instances.0.commandClasses.32.data.level = 255 (0x000000ff)
[2016-10-22 18:38:10.958] [D] [zway] RECEIVED: ( 01 08 00 04 00 18 02 84 07 6A )
[2016-10-22 18:38:10.959] [D] [zway] SENT ACK
[2016-10-22 18:38:10.959] [D] [zway] SETDATA devices.24.data.lastReceived = 0 (0x00000000)
[2016-10-22 18:38:10.959] [I] [zway] Node 24:0 CC Wakeup: Wakeup notification
[2016-10-22 18:38:10.959] [D] [zway] SETDATA devices.24.instances.0.commandClasses.132.data.lastWakeup = 1477154290 (0x580b95f2)
[2016-10-22 18:38:10.959] [D] [zway] SETDATA devices.24.data.isAwake = True
[2016-10-22 18:38:10.959] [D] [zway] SETDATA devices.24.data.lastSend = 8406369 (0x00804561)
[2016-10-22 18:38:10.959] [D] [zway] SETDATA devices.24.data.lastNonceGet = 8405369 (0x00804179)
[2016-10-22 18:38:10.960] [D] [zway] Running wakeup handler for node 24
[2016-10-22 18:38:10.960] [D] [zway] SETDATA devices.24.instances.0.commandClasses.132.data.lastSleep = 1477154290 (0x580b95f2)
[2016-10-22 18:38:10.960] [I] [zway] Node 24:0 CC Wakeup: Send node to sleep
[2016-10-22 18:38:10.960] [I] [zway] Adding job: Wakeup Sleep
[2016-10-22 18:38:11.029] [D] [zway] SENDING (cb 0xc4): ( 01 09 00 13 18 02 84 08 05 C4 B2 )
[2016-10-22 18:38:11.031] [D] [zway] RECEIVED ACK
[2016-10-22 18:38:11.035] [D] [zway] RECEIVED: ( 01 04 01 13 01 E8 )
[2016-10-22 18:38:11.035] [D] [zway] SENT ACK
[2016-10-22 18:38:11.035] [D] [zway] Delivered to Z-Wave stack
[2016-10-22 18:38:11.051] [D] [zway] RECEIVED: ( 01 07 00 13 C4 00 00 01 2E )
[2016-10-22 18:38:11.051] [D] [zway] SENT ACK
[2016-10-22 18:38:11.051] [I] [zway] Job 0x13 (Wakeup Sleep): Delivered
[2016-10-22 18:38:11.051] [D] [zway] SETDATA devices.24.data.lastPacketInfo.delivered = True
[2016-10-22 18:38:11.051] [D] [zway] SETDATA devices.24.data.lastPacketInfo.packetLength = 5 (0x00000005)
[2016-10-22 18:38:11.051] [D] [zway] SETDATA devices.24.data.lastPacketInfo.deliveryTime = 15 (0x0000000f)
[2016-10-22 18:38:11.051] [D] [zway] SETDATA devices.24.data.lastPacketInfo = **********
[2016-10-22 18:38:11.051] [D] [zway] SendData Response with callback 0xc4 received: received by recipient
[2016-10-22 18:38:11.052] [D] [zway] SETDATA devices.24.data.lastSend = 8406378 (0x0080456a)
[2016-10-22 18:38:11.052] [D] [zway] Job 0x13 (Wakeup Sleep): success
[2016-10-22 18:38:11.052] [D] [zway] SETDATA devices.24.data.isAwake = False
[2016-10-22 18:38:11.052] [D] [zway] Sending the queue for node 24 into sleep
[2016-10-22 18:38:11.052] [I] [zway] Removing job: Wakeup Sleep

**Comment by PoltoS** [comment]
The control command reached Z-Way. Sure you have Trap Events module on and ready to autogenerate vDevs?

**Comment by bogri** [comment]
Aah, ok, you're right. I didn't have the Trap Events module. I've enabled it and got one button; vDev in the UI. But I was expecting to get at least two (even more since the device has support for double press, and long press). But the reporting from the device, when pressing the "full circle" button vs the + button looks like:

**Pressing the full circle button**
[2016-10-23 00:30:08.942] [D] [zway] RECEIVED: ( 01 09 00 04 00 18 03 20 01 FF 37 )
[2016-10-23 00:30:08.942] [D] [zway] SENT ACK
[2016-10-23 00:30:08.942] [D] [zway] SETDATA devices.24.data.lastReceived = 0 (0x00000000)
[2016-10-23 00:30:08.943] [D] [zway] SETDATA devices.1.instances.0.commandClasses.32.data.srcNodeId = 24 (0x00000018)
[2016-10-23 00:30:08.943] [D] [zway] SETDATA devices.1.instances.0.commandClasses.32.data.srcInstanceId = 0 (0x00000000)
[2016-10-23 00:30:08.943] [D] [zway] SETDATA devices.1.instances.0.commandClasses.32.data.level = 255 (0x000000ff)
[2016-10-23 00:30:08.986] [I] [core] Notification: device-info (device-OnOff): {"dev":" (24.0.0) Button","l":"on"}
[2016-10-23 00:30:09.029] [I] [core] [BaseModule-70] Set lastLevel to on for ZWayVDev_zway_Remote_24-0-0-B (was off)
[2016-10-23 00:30:09.267] [D] [zway] RECEIVED: ( 01 08 00 04 00 18 02 84 07 6A )
[2016-10-23 00:30:09.267] [D] [zway] SENT ACK
[2016-10-23 00:30:09.267] [D] [zway] SETDATA devices.24.data.lastReceived = 0 (0x00000000)
[2016-10-23 00:30:09.267] [I] [zway] Node 24:0 CC Wakeup: Wakeup notification
[2016-10-23 00:30:09.268] [D] [zway] SETDATA devices.24.instances.0.commandClasses.132.data.lastWakeup = 1477175409 (0x580be871)
[2016-10-23 00:30:09.268] [D] [zway] SETDATA devices.24.data.isAwake = True
[2016-10-23 00:30:09.268] [D] [zway] SETDATA devices.24.data.lastSend = 10484007 (0x009ff927)
[2016-10-23 00:30:09.268] [D] [zway] SETDATA devices.24.data.lastNonceGet = 10483007 (0x009ff53f)
[2016-10-23 00:30:09.268] [D] [zway] Running wakeup handler for node 24
[2016-10-23 00:30:09.268] [D] [zway] SETDATA devices.24.instances.0.commandClasses.132.data.lastSleep = 1477175409 (0x580be871)
[2016-10-23 00:30:09.268] [I] [zway] Node 24:0 CC Wakeup: Send node to sleep
[2016-10-23 00:30:09.269] [I] [zway] Adding job: Wakeup Sleep
[2016-10-23 00:30:09.338] [D] [zway] SENDING (cb 0xd7): ( 01 09 00 13 18 02 84 08 05 D7 A1 )
[2016-10-23 00:30:09.340] [D] [zway] RECEIVED ACK
[2016-10-23 00:30:09.344] [D] [zway] RECEIVED: ( 01 04 01 13 01 E8 )
[2016-10-23 00:30:09.344] [D] [zway] SENT ACK
[2016-10-23 00:30:09.344] [D] [zway] Delivered to Z-Wave stack
[2016-10-23 00:30:09.358] [D] [zway] RECEIVED: ( 01 07 00 13 D7 00 00 01 3D )
[2016-10-23 00:30:09.359] [D] [zway] SENT ACK
[2016-10-23 00:30:09.359] [I] [zway] Job 0x13 (Wakeup Sleep): Delivered
[2016-10-23 00:30:09.359] [D] [zway] SETDATA devices.24.data.lastPacketInfo.delivered = True
[2016-10-23 00:30:09.359] [D] [zway] SETDATA devices.24.data.lastPacketInfo.packetLength = 5 (0x00000005)
[2016-10-23 00:30:09.359] [D] [zway] SETDATA devices.24.data.lastPacketInfo.deliveryTime = 14 (0x0000000e)
[2016-10-23 00:30:09.360] [D] [zway] SETDATA devices.24.data.lastPacketInfo = **********
[2016-10-23 00:30:09.360] [D] [zway] SendData Response with callback 0xd7 received: received by recipient
[2016-10-23 00:30:09.360] [D] [zway] SETDATA devices.24.data.lastSend = 10484016 (0x009ff930)
[2016-10-23 00:30:09.360] [D] [zway] Job 0x13 (Wakeup Sleep): success
[2016-10-23 00:30:09.360] [D] [zway] SETDATA devices.24.data.isAwake = False
[2016-10-23 00:30:09.360] [D] [zway] Sending the queue for node 24 into sleep
[2016-10-23 00:30:09.360] [I] [zway] Removing job: Wakeup Sleep

**Pressing the + button**
[2016-10-23 00:30:12.626] [D] [zway] RECEIVED: ( 01 09 00 04 00 18 03 20 01 FF 37 )
[2016-10-23 00:30:12.626] [D] [zway] SENT ACK
[2016-10-23 00:30:12.626] [D] [zway] SETDATA devices.24.data.lastReceived = 0 (0x00000000)
[2016-10-23 00:30:12.626] [D] [zway] SETDATA devices.1.instances.0.commandClasses.32.data.srcNodeId = 24 (0x00000018)
[2016-10-23 00:30:12.626] [D] [zway] SETDATA devices.1.instances.0.commandClasses.32.data.srcInstanceId = 0 (0x00000000)
[2016-10-23 00:30:12.626] [D] [zway] SETDATA devices.1.instances.0.commandClasses.32.data.level = 255 (0x000000ff)
[2016-10-23 00:30:12.636] [I] [core] Notification: device-info (device-OnOff): {"dev":" (24.0.0) Button","l":"on"}
[2016-10-23 00:30:12.951] [D] [zway] RECEIVED: ( 01 08 00 04 00 18 02 84 07 6A )
[2016-10-23 00:30:12.951] [D] [zway] SENT ACK
[2016-10-23 00:30:12.951] [D] [zway] SETDATA devices.24.data.lastReceived = 0 (0x00000000)
[2016-10-23 00:30:12.951] [I] [zway] Node 24:0 CC Wakeup: Wakeup notification
[2016-10-23 00:30:12.951] [D] [zway] SETDATA devices.24.instances.0.commandClasses.132.data.lastWakeup = 1477175412 (0x580be874)
[2016-10-23 00:30:12.951] [D] [zway] SETDATA devices.24.data.isAwake = True
[2016-10-23 00:30:12.951] [D] [zway] SETDATA devices.24.data.lastSend = 10484370 (0x009ffa92)
[2016-10-23 00:30:12.952] [D] [zway] SETDATA devices.24.data.lastNonceGet = 10483370 (0x009ff6aa)
[2016-10-23 00:30:12.952] [D] [zway] Running wakeup handler for node 24
[2016-10-23 00:30:12.952] [D] [zway] SETDATA devices.24.instances.0.commandClasses.132.data.lastSleep = 1477175412 (0x580be874)
[2016-10-23 00:30:12.952] [I] [zway] Node 24:0 CC Wakeup: Send node to sleep
[2016-10-23 00:30:12.952] [I] [zway] Adding job: Wakeup Sleep
[2016-10-23 00:30:13.022] [D] [zway] SENDING (cb 0xd8): ( 01 09 00 13 18 02 84 08 05 D8 AE )
[2016-10-23 00:30:13.024] [D] [zway] RECEIVED ACK
[2016-10-23 00:30:13.027] [D] [zway] RECEIVED: ( 01 04 01 13 01 E8 )
[2016-10-23 00:30:13.027] [D] [zway] SENT ACK
[2016-10-23 00:30:13.027] [D] [zway] Delivered to Z-Wave stack
[2016-10-23 00:30:13.042] [D] [zway] RECEIVED: ( 01 07 00 13 D8 00 00 02 31 )
[2016-10-23 00:30:13.042] [D] [zway] SENT ACK
[2016-10-23 00:30:13.042] [I] [zway] Job 0x13 (Wakeup Sleep): Delivered
[2016-10-23 00:30:13.042] [D] [zway] SETDATA devices.24.data.lastPacketInfo.delivered = True
[2016-10-23 00:30:13.043] [D] [zway] SETDATA devices.24.data.lastPacketInfo.packetLength = 5 (0x00000005)
[2016-10-23 00:30:13.043] [D] [zway] SETDATA devices.24.data.lastPacketInfo.deliveryTime = 14 (0x0000000e)
[2016-10-23 00:30:13.043] [D] [zway] SETDATA devices.24.data.lastPacketInfo = **********
[2016-10-23 00:30:13.043] [D] [zway] SendData Response with callback 0xd8 received: received by recipient
[2016-10-23 00:30:13.043] [D] [zway] SETDATA devices.24.data.lastSend = 10484379 (0x009ffa9b)
[2016-10-23 00:30:13.043] [D] [zway] Job 0x13 (Wakeup Sleep): success
[2016-10-23 00:30:13.043] [D] [zway] SETDATA devices.24.data.isAwake = False
[2016-10-23 00:30:13.043] [D] [zway] Sending the queue for node 24 into sleep
[2016-10-23 00:30:13.043] [I] [zway] Removing job: Wakeup Sleep

Any ideas why I don't see (at least) two buttons; vDevs?

**Comment by PoltoS** [comment]
Because it sends same commands on both presses. You should either enwable scene mode (if supported) or change association from 1 to 1:1 and 1:2 correspondingly for two buttons (should be group 2 and 3 I believe, since group 1 is LifeLine).

In both cases two different commands will come from these buttons and Z-Way will be able to distinguish them

**Comment by bogri** [comment]
Yes that was it, thanx a lot for the tip. If I configure every button at scene mode I get four **scene buttons** (24.0.0.10, 20, 30, 40) Button). I had the config at duo mode, but I still would have expected that I would get two **switch buttons**, and not just one. But, as you say, if it's the same command that is sent, there's not much the server can do about it :/

**Comment by bogri** [comment]
One more question. How can I remove vDevs that are generated by the Trap Events module but no longer used? The device has two different modes "Central Scene" and "Scene Activation". Now when I switched configuration from Scene Activation to Central Scene I got 4 more buttons (vDevs: (24.0.0.1, 2, 3, 4) Button) and the old ones ((24.0.0.10, 20, 30, 40) Button) are not longer relevant since all events from the device goes to the 1, 2, 3, 4 vDevs; e.g.

[2016-10-23 18:08:06.872] [I] [core] Notification: device-info (device-OnOff): {"dev":" (24.0.0.1) Button","l":"on"}

So, how can I remove the old generated vDevs?
